src/bcf-toolkit/Converter/Bcf30/SchemaConverterToBcf21.cs

```diff
@@ -1,11 +1,296 @@
+      .AddReferenceLinks(from.Topic.ReferenceLinks.ToList())
+      .SetTitle(from.Topic.Title)
+      .SetPriority(from.Topic.Priority)
+      .SetIndex(from.Topic.Index)
+      .AddLabels(from.Topic.Labels.ToList())
+      .SetCreationDate(from.Topic.CreationDate)
+      .SetCreationAuthor(from.Topic.CreationAuthor)
+      .SetModifiedDate(from.Topic.ModifiedDate)
+      .SetModifiedAuthor(from.Topic.ModifiedAuthor)
+      .SetDueDate(from.Topic.DueDate)
+      .SetAssignedTo(from.Topic.AssignedTo)
+      .SetDescription(from.Topic.Description)
+      .SetStage(from.Topic.Stage)
+      .SetBimSnippet(ConvertBimSnippet(from.Topic.BimSnippet))
+      .AddDocumentReferences(from.Topic.DocumentReferences
+        .Select(ConvertDocumentReference).ToList())
+      .AddComments(from.Topic.Comments.Select(ConvertComment).ToList())
+      .AddViewPoints(from.Topic.Viewpoints.Select(ConvertViewPoint).ToList())
+      .AddRelatedTopics(from.Topic.RelatedTopics.Select(ConvertRelatedTopic).ToList());
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.HeaderFile ConvertHeaderFile(Model.Bcf30.File from) {
+    var builder = new HeaderFileBuilder();
+    builder
+      .SetIfcProject(from.IfcProject)
+      .SetIfcSpatialStructureElement(from.IfcSpatialStructureElement)
+      .SetIsExternal(from.IsExternal)
+      .SetFileName(from.Filename)
+      .SetDate(from.Date)
+      .SetReference(from.Reference);
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.BimSnippet? ConvertBimSnippet(Model.Bcf30.BimSnippet? from) {
+    if (from is null) {
+      return null;
+    }
+
+    var builder = new BimSnippetBuilder();
+    builder
+      .SetSnippetType(from.SnippetType)
+      .SetIsExternal(from.IsExternal)
+      .SetReference(from.Reference)
+      .SetReferenceSchema(from.ReferenceSchema);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.TopicDocumentReference ConvertDocumentReference(
+    Model.Bcf30.DocumentReference from) {
+    var builder = new DocumentReferenceBuilder();
+    var isExternal = string.IsNullOrEmpty(from.Url);
+    builder
+      .SetGuid(from.Guid)
+      .SetDescription(from.Description)
+      .SetIsExternal(isExternal)
+      .SetReferencedDocument(isExternal ? from.Url : from.DocumentGuid);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.Comment ConvertComment(Model.Bcf30.Comment from) {
+    var builder = new CommentBuilder();
+
+    builder
+      .SetGuid(from.Guid)
+      .SetDate(from.Date)
+      .SetAuthor(from.Author)
+      .SetCommentProperty(from.CommentProperty)
+      .SetViewPointGuid(from.Viewpoint.Guid)
+      .SetModifiedDate(from.ModifiedDate)
+      .SetModifiedAuthor(from.ModifiedAuthor);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.ViewPoint ConvertViewPoint(
+    Model.Bcf30.ViewPoint from) {
+
+    var viewPoint = new ViewPoint();
+
+    viewPoint.VisualizationInfo =
```

Comment:
Maybe a builder would be nice to use. There are fields that are not filled:
* Viewpoint
* Snapshot
* Index
* Guid

Reply:
Done.